allennlp/data/instance.py

```diff
@@ -111,3 +113,34 @@ def duplicate(self) -> "Instance":
         new = Instance({k: field.duplicate() for k, field in self.fields.items()})
         new.indexed = self.indexed
         return new
+
+    def human_readable_dict(self) -> JsonDict:
+        """
+        This function help to save formated instances to json files or print for human readability.
+        Use case includes example-based explanation, where it's better to have a output file or
+        rather than printing or logging.
+
+        For example, - if the field is LabelField, then we just output, {"label": `field.label` (string)}
+                     - if the field is TextField, then we just output,
+                     {"tokens": `field.tokens` (preferrably un-numericalized tokens)}
+                     - the instances output then is {"label": `field.label`, "tokens": `field.tokens`}
+
+        If the field's key in the instance is different from the field's default, we will use the
+        key in the instance rather than the field's default.
+
+        For example, - if instance is {"hahah": `TextField`}
+                     - output from this function will be {"hahah": `field.tokens`} rather than
+                     {"tokens" (`TextField` default key): `field.tokens`}
+
+        """
+        ret = {}
+        for key, field in self.fields.items():
+            human_readable_dict = field.human_readable_dict()
+            assert len(human_readable_dict) == 1  # each field (as dict) has its default key
+            key_in_dict = list(human_readable_dict.keys())[0]
+            if key != key_in_dict:
+                human_readable_dict[key] = deepcopy(human_readable_dict[key_in_dict])
+                del human_readable_dict[key_in_dict]
+            ret.update(human_readable_dict)
+
+        return ret
```

Comment:
I think it would be better to keep it simple and nested by the field name:

```suggestion
        return {key: field.human_readable_dict() for key, field in self.fields.items()}
```

Reply:
However, wouldn't writing methods this way will make us end up with something like `{"tokens": {"tokens": ["I", "like", "apple"]}}`?

Reply:
How about we rename the function to be `human_readable_repr`?

Reply:
> How about we rename the function to be human_readable_repr?

That's a good idea. That way `TextField.human_readable_repr()` could just return `List[str]`, right? Is that what you're thinking?

Reply:
Exactly. But now, what's the expected type from the base class? `object`? hhh